app/frontend/actions/delegate.ts

```diff
@@ -34,9 +36,13 @@ export default (store: Store) => {
     if (hasPoolIdentifiersChanged(state)) {
       return
     }
-    const poolInfo = !state.shelleyDelegation?.selectedPool?.name
-      ? await getWallet().getPoolInfo(state.shelleyDelegation?.selectedPool?.url)
-      : {}
+    const poolInfo = (!state.shelleyDelegation?.selectedPool?.name &&
```

Comment:
This overriding of potentially undefined values with empty ones seems a bit fragile and I'd have a hard time reasoning about the possible consequences of it. I think it can be rewritten more sensibly like this:

```javascript
    const selectedPool = state.shelleyDelegation?.selectedPool
    const poolInfo = (!selectedPool?.name && selectedPool?.url) ?
      (await getWallet()?.getPoolInfo(selectedPool?.url))
      : null

    if (hasPoolIdentifiersChanged(state)) {
      return
    }
    const newState = getState()
    setState({
      shelleyDelegation: {
        ...state.shelleyDelegation,
        selectedPool: selectedPool ? {
          ...selectedPool,
          ...poolInfo,
        } : null,
```

@PeterBenc please check if I'm not missing anything by any chance as I think you were originally working on this

Reply:
Yeah I agree, we shouldn't overide the values, that handled in `stakePoolInfo.tsx` file, What @refi93 proposes should imho work as expected so I am curious what was the reason to change the previous behavior, the difference being only doing `...null` instead of `...{}` which should be the same.

Reply:
The problem was that type for `Stakepool` doesn't have any optional properties. As it was written if the poolInfo or selectedPool can't be fetched or doesn't exist then you end up with an empty object. I was thinking it would be better to have defaults than to type all the properties as optional. 

I am updating with the solution proposed by @refi93 because it does make more sense